tidy3d/components/data/dataset.py

```diff
@@ -32,6 +32,15 @@ class AbstractFieldDataset(Dataset, ABC):
     def field_components(self) -> Dict[str, DataArray]:
         """Maps the field components to thier associated data."""
 
+    def apply_phase(self, phase: float) -> AbstractFieldDataset:
+        """Create a copy of this dataset where all elements are phase shifted by value (radians)."""
+        phasor = np.exp(1j * phase)
+        field_components_shifted = {}
+        for fld_name, fld_cmp in self.field_components.items():
+            fld_cmp_shifted = phasor * fld_cmp
+            field_components_shifted[fld_name] = fld_cmp_shifted
+        return self.updated_copy(**field_components_shifted)
```

Comment:
Should we avoid creating a copy here if `phase == 0.0`? This will be run for every plot, most times with the default value, and I wander if it would have noticeable impact for large datasets.

Reply:
sure, I can just return. Note that in the `SimulationData.plot_field()`, I have a [check](https://github.com/flexcompute/tidy3d/blob/tyler/phase/tidy3d/components/data/sim_data.py#L447-L455) that phase != 0.0 before calling `apply_phase`, so this shouldn't be called in the plotting unless the phase is non-zero. However, I'll add an extra check here just in case.